agent/pkg/providers/user_role_provider.go

```diff
@@ -0,0 +1,180 @@
+package providers
```

Comment:
so we won't end up with a package with too many providers, I suggest start splitting them:
"user role provider"
"user provide"